--- a/crypto/aes/aes_core.c
+++ b/crypto/aes/aes_core.c
@@ -52,9 +52,10 @@
 
 #if defined(OPENSSL_AES_CONST_TIME) && !defined(AES_ASM)
 
-# if (defined(_WIN32) || defined(_WIN64)) && !defined(__MINGW32__)
+# if (defined(_WIN32) || defined(_WIN64)) && !defined(__MINGW32__) \
+     && !defined(__CYGWIN__)
 #  define U64(C) C##UI64
-# elif defined(__arch64__)
+# elif defined(__arch64__) || defined(__aarch64__)
 #  define U64(C) C##UL
 # else
 #  define U64(C) C##ULL
--- a/crypto/aes/aes_x86core.c
+++ b/crypto/aes/aes_x86core.c
@@ -80,9 +80,10 @@
 #undef GETU32
 #define GETU32(p) (*((u32*)(p)))
 
-#if (defined(_WIN32) || defined(_WIN64)) && !defined(__MINGW32__)
+#if (defined(_WIN32) || defined(_WIN64)) && !defined(__MINGW32__) \
+    && !defined(__CYGWIN__)
 #define U64(C)  C##UI64
-#elif defined(__arch64__)
+#elif defined(__arch64__) || defined(__aarch64__)
 #define U64(C)  C##UL
 #else
 #define U64(C)  C##ULL
--- a/crypto/bio/bio_addr.c
+++ b/crypto/bio/bio_addr.c
@@ -799,7 +799,7 @@ static int addr_strings(const BIO_ADDR *ap, int numeric,
 
         struct servent *se;
         /* Apparently, on WIN64, s_proto and s_port have traded places... */
-#ifdef _WIN64
+#if defined(_WIN64) && !defined(__CYGWIN__)
         struct servent se_fallback = { NULL, NULL, NULL, 0 };
 #else
         struct servent se_fallback = { NULL, NULL, 0, NULL };
--- a/crypto/bn/bn_srp.c
+++ b/crypto/bn/bn_srp.c
@@ -16,9 +16,10 @@
 #include "crypto/bn_srp.h"
 
 # if (BN_BYTES == 8)
-#  if (defined(_WIN32) || defined(_WIN64)) && !defined(__MINGW32__)
+#  if (defined(_WIN32) || defined(_WIN64)) && !defined(__MINGW32__) \
+      && !defined(__CYGWIN__)
 #   define bn_pack4(a1,a2,a3,a4) ((a1##UI64<<48)|(a2##UI64<<32)|(a3##UI64<<16)|a4##UI64)
-#  elif defined(__arch64__)
+#  elif defined(__arch64__) || defined(__aarch64__)
 #   define bn_pack4(a1,a2,a3,a4) ((a1##UL<<48)|(a2##UL<<32)|(a3##UL<<16)|a4##UL)
 #  else
 #   define bn_pack4(a1,a2,a3,a4) ((a1##ULL<<48)|(a2##ULL<<32)|(a3##ULL<<16)|a4##ULL)
--- a/crypto/sha/sha512.c
+++ b/crypto/sha/sha512.c
@@ -67,9 +67,10 @@
 # define SHA512_BLOCK_CAN_MANAGE_UNALIGNED_DATA
 #endif
 
-#if (defined(_WIN32) || defined(_WIN64)) && !defined(__MINGW32__)
+#if (defined(_WIN32) || defined(_WIN64)) && !defined(__MINGW32__) \
+    && !defined(__CYGWIN__)
 # define U64(C)     C##UI64
-#elif defined(__arch64__)
+#elif defined(__arch64__) || defined(__aarch64__)
 # define U64(C)     C##UL
 #else
 # define U64(C)     C##ULL
--- a/crypto/whrlpool/wp_block.c
+++ b/crypto/whrlpool/wp_block.c
@@ -47,9 +47,10 @@
 #include <string.h>
 
 typedef unsigned char u8;
-#if (defined(_WIN32) || defined(_WIN64)) && !defined(__MINGW32)
+#if (defined(_WIN32) || defined(_WIN64)) && !defined(__MINGW32) \
+    && !defined(__CYGWIN__)
 typedef unsigned __int64 u64;
-#elif defined(__arch64__)
+#elif defined(__arch64__) || defined(__aarch64__)
 typedef unsigned long u64;
 #else
 typedef unsigned long long u64;
--- a/include/crypto/modes.h
+++ b/include/crypto/modes.h
@@ -10,11 +10,12 @@
 /* This header can move into provider when legacy support is removed */
 #include <openssl/modes.h>
 
-#if (defined(_WIN32) || defined(_WIN64)) && !defined(__MINGW32__)
+#if (defined(_WIN32) || defined(_WIN64)) && !defined(__MINGW32__) \
+    && !defined(__CYGWIN__)
 typedef __int64 i64;
 typedef unsigned __int64 u64;
 # define U64(C) C##UI64
-#elif defined(__arch64__)
+#elif defined(__arch64__) || defined(__aarch64__)
 typedef long i64;
 typedef unsigned long u64;
 # define U64(C) C##UL
--- a/include/openssl/sha.h
+++ b/include/openssl/sha.h
@@ -99,9 +99,10 @@ extern "C" {
  * wide big-endian values.
  */
 #  define SHA512_CBLOCK   (SHA_LBLOCK*8)
-#  if (defined(_WIN32) || defined(_WIN64)) && !defined(__MINGW32__)
+#  if (defined(_WIN32) || defined(_WIN64)) && !defined(__MINGW32__) \
+      && !defined(__CYGWIN__)
 #   define SHA_LONG64 unsigned __int64
-#  elif defined(__arch64__)
+#  elif defined(__arch64__) || defined(__aarch64__)
 #   define SHA_LONG64 unsigned long
 #  else
 #   define SHA_LONG64 unsigned long long
